### PR TITLE
Ignore nulls in partitions to be processed

### LINKED ISSUE #544: [CT-2136] [Bug] Null partitions fail to merge on insert_overwrite materialization strategy when using _dbt_max_partition
### Is this a new bug in dbt-bigquery?

- [X] I believe this is a new bug in dbt-bigquery
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I have seen a few reports recently of incremental `insert_overwrite` with `_dbt_max_partition` causing null partitions somehow. Which is itself, a separate issue, but the fact there are nulls causes the merge statement to fail:
Example: https://github.com/Velir/dbt-ga4/issues/128
```
21:43:49  Database Error in model fct_ga4__sessions_daily (models/marts/core/fct_ga4__sessions_daily.sql)
  Query error: Array cannot have a null element._field_1 at [76:46]
```

Other semi-related example: https://github.com/dbt-labs/dbt-bigquery/pull/285#issuecomment-1427137218

### Expected Behavior

partitions created / found without nulls
merge happens successfully

We can prevent the error by overwriting the lines in the `insert_overwrite` macro:
https://github.com/dbt-labs/dbt-bigquery/blob/a3d70544ab0cb736676c17932b8f52cd215fe1d7/dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql
 
changing lines 134-139 from:
``` 
     -- 2. define partitions to update
      set (dbt_partitions_for_replacement) = (
          select as struct
              array_agg(distinct {{ partition_by.render_wrapped() }})
          from {{ tmp_relation }}
      );
 ```
to:
```
      -- 2. define partitions to update
      set (dbt_partitions_for_replacement) = (
          select as struct
              array_agg(distinct {{ partition_by.render_wrapped() }} IGNORE NULLS)
          from {{ tmp_relation }}
      );
 ```

### Steps To Reproduce

I believe this creates the error:

in dbt Cloud, run the incremental twice in a row, -or- force it somehow to return no new rows. I think no new rows is creating a null partition

The `_dbt_max_partition` [does](https://github.com/dbt-labs/dbt-bigquery/blob/a3d70544ab0cb736676c17932b8f52cd215fe1d7/dbt/include/bigquery/macros/materializations/incremental_strategy/common.sql#L17) seem to remove nulls, but [this](https://github.com/dbt-labs/dbt-bigquery/blob/a3d70544ab0cb736676c17932b8f52cd215fe1d7/dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql#L134-L138) does not:
```
      -- 2. define partitions to update
      set (dbt_partitions_for_replacement) = (
          select as struct
              array_agg(distinct {{ partition_by.render_wrapped() }})
          from {{ tmp_relation }}
```
causing the `array_agg` function to error out

### Relevant log output

```shell
See here: https://github.com/Velir/dbt-ga4/issues/128
```

### Environment

```markdown
- OS:
- Python:
- dbt-core:1.4.1
- dbt-bigquery:
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Fleid**
resolves #544

Missing tests :/

### CODE REVIEW

**Comment by Fleid** [comment]
Missing tests, can't find the time to write them

**Comment by VersusFacit** [comment]
Chiming in that I don't even think we can add tests given everything I've been researching. I'm waiting on sign off from people higher up the  engineering stack from me though before I overzealously merge.

**Review by VersusFacit — APPROVED** [review]
Team has agreed to merge this fix as a stopgap, but we want to get to the bottom of things.

 I'm gonna leave a note on the relevant issue cards saying as much

**Pull request merged**
